### ISSUE TITLE
Broken Birt Report Engine

### ISSUE THREAD

**Issue by ioagel** [eclipse-platform/eclipse.platform#171]
Birt 4.9.0 Report Engine API integrated into my app, is broken with error:
`java.lang.SecurityException: class "org.eclipse.core.runtime.Platform"'s signer information does not match signer information of other classes in the same package`

I had to use the latest maven dependency for `org.eclipse.core.runtime`, to fix it:
```xml
<dependency>
    <groupId>org.eclipse.platform</groupId>
    <artifactId>org.eclipse.core.runtime</artifactId>
    <version>3.26.0</version>
</dependency>
```

How is it possible using a specific version like `3.25.0` for a couple of months now, working without any problems, to be broken by a new release?

**Comment by laeubi** [comment]
Platform sadly uses (too broad) version ranges for their poms when autogenerating the pom.xml, this can pull in newer stuff if a release is made.

**Comment by merks** [comment]
This problem existed in the 4.24 release and is fixed in the 4.25 release.  It was reported here:

https://github.com/eclipse-equinox/equinox/issues/39

Other instances of the problem were also found:

https://github.com/eclipse-platform/eclipse.platform.ui/issues/225#issuecomment-1192344077

But those too are all fixes in the latest release.

I cannot answer why you didn't have a problem before, that's a detail of how Maven resolves a dependency given a version range.  But, as you say,  it's fixed by using the latest versions.
